src/aggregate/expr/parser.y

```diff
@@ -42,12 +42,18 @@
 
 }
 
-%syntax_error {  
+%syntax_error {
 
-    rm_asprintf(&ctx->errorMsg, "Syntax error at offset %d near '%.*s'", TOKEN.pos, TOKEN.len, TOKEN.s);
+    if (ctx->errorMsg) {
+        char *reason = ctx->errorMsg;
+        rm_asprintf(&ctx->errorMsg, "Syntax error at offset %d near '%.*s': %s", TOKEN.pos, TOKEN.len, TOKEN.s, reason);
+        rm_free(reason);
```

Comment:
Can also `ctx->errorMsg = NULL;` ?

Reply:
Are you asking if it could be or should we do it?
Is it possible? it might, be hard to tell. there are certainly places where it was not NULL (and caused the leak).
Should we do it here? no. This is how the error message is passed out

Reply:
Sorry for not being clear.
At this point we know `errorMsg` is not `NULL`, we report it, free it, and I suggest we also set the member to `NULL`

Reply:
OK, cannot set it to `NULL` (freeing only the previous string , keeping the new string using `rm_asprintf`)